geniza/corpus/models.py

```diff
@@ -666,8 +669,38 @@ def iiif_images(self, filter_side=False):
                     or i in b.selected_images
                 ]
 
+        # order returned images according to image_order_override, if present
+        if self.image_order_override:
+            unordered_images = iiif_images
+            iiif_images = []
+            for canvas in self.image_order_override:
+                matches = [
+                    idx
+                    for idx, img in enumerate(unordered_images)
+                    if img["canvas"] == canvas
+                ]
+                if matches:
+                    iiif_images.append(unordered_images.pop(matches[0]))
```

Comment:
What about making a dict of images keyed on canvas uri, and then adding them to the list based on uris in override order? You can `pop` values from a dict, so then you could still add the remaining canvases.

Could maybe even do this when initially constructing the list of images from fragments — create a dict with canvas uris as keys, return values if no order override (and dict order preserved as of python 3.7 so it should do what we want). Retrieving by uri would need to handle missing fragments in the case of filtering non-document images

Reply:
Refactored along those lines, let me know if I missed anything!